### ISSUE TITLE
Cannot pass `name` prop to `<Search>`

### ISSUE THREAD

**Issue by s3812497** [carbon-design-system/carbon-components-svelte#1520]
Currently, the `name` prop cannot be passed to the `<input>` element in the `<Search>` component.
This prevents it from being usable with forms, to append a search param to the action url.

e.g. This navigates to `/search?q=value`
```html
<form action="/search">
  <input type="search" name="q" />
  <button>Search</button>
</form>
```

Without the `name` prop, it can only ever navigate to `/search?`
```html
<form action="/search">
  <input type="search" />
  <button>Search</button>
</form>
```

**Comment by metonym** [comment]
`$$restProps` are forwarded to the `input` element in `Search`: https://github.com/carbon-design-system/carbon-components-svelte/blob/c6373dc9b6db53a73784e8e480d8f6e2bc278948/src/Search/Search.svelte#L131

I wasn't able to reproduce this with the following code. Submitting the form navigated to `/search?q=<query>`. Inspecting the DOM showed that the `name` attribute was correctly set on the `input` element.

```svelte
<script>
  import { Search } from "carbon-components-svelte";

  let value = "";
</script>

<form action="/search">
  <Search bind:value name="q" />
  <button>Search</button>
</form>
```

**Comment by s3812497** [comment]
Whoops, I guess it's just a typescript error thing.

Besides that, it might be useful to pass it the prop manually as the Svelte docs state $$restProps shouldn't be relied on (I personally haven't used it due to this but it looks useful)

> $$restProps contains only the props which are not declared with export. It can be used to pass down other unknown attributes to an element in a component. It shares the same optimisation problems as $$props, and is likewise not recommended.

https://svelte.dev/docs#template-syntax-attributes-and-props

**Comment by metonym** [comment]
Thanks for flagging this. Fixed in [v0.70.12](https://github.com/carbon-design-system/carbon-components-svelte/releases/tag/v0.70.12)
